src/parquet/column/properties.h

```diff
@@ -174,9 +219,14 @@ class PARQUET_EXPORT WriterProperties {
     // Encoding used for each column if not a specialized one is defined as
     // part of encodings_
     Encoding::type default_encoding_;
+    Encoding::type default_dictionary_encoding_;
+    Encoding::type default_level_encoding_;
     std::unordered_map<std::string, Encoding::type> encodings_;
+    std::unordered_map<std::string, Encoding::type> dictionary_encodings_;
+    std::unordered_map<std::string, Encoding::type> level_encodings_;
```

Comment:
This proliferation of `unordered_map`s is worrying from performance point of view as multiple repeated hash calculations would be required for each column.